Lib/datetime.py

```diff
@@ -884,6 +884,40 @@ def fromisoformat(cls, date_string):
         except Exception:
             raise ValueError(f'Invalid isoformat string: {date_string!r}')
 
+    @classmethod
+    def fromisocalendar(cls, year, week, day):
+        """Construct a date from the ISO year, week number and weekday.
+
+        This is the inverse of the date.isocalendar() function"""
+        # Year is bounded this way because 9999-12-31 is (9999, 52, 5)
+        if not 0 < year < 10000:
```

Comment:
MINYEAR/MAXYEAR should be used here, no?

Reply:
Yes, good point.